src/liquid/components/ld-notification/ld-notification.tsx

```diff
@@ -0,0 +1,190 @@
+    )
+    if (firstErrorNotificationIndex === -1) {
+      this.queue = [...this.queue, newNotification]
+      this.currentNotification = newNotification
+      return
+    }
+
+    this.queue.splice(firstErrorNotificationIndex, 0, newNotification)
+    this.queue = [...this.queue]
+  }
+
+  @Listen('ldNotificationDismiss', {
+    target: 'window',
+    passive: true,
+  })
+  handleNotificationDismiss() {
+    if (!this.currentNotification) return
+
+    this.queueDismissed.unshift(this.queue.pop())
+    this.queueDismissed = [...this.queueDismissed]
+    this.queue = [...this.queue]
+    this.currentNotification = this.queue[this.queue.length - 1]
+
+    this.fadeoutTimeouts.push(
+      setTimeout(() => {
+        this.queueDismissed = this.queueDismissed.slice(0, -1)
+      }, FADE_TRANSITION_DURATION)
+    )
+  }
+
+  @Listen('ldNotificationClear', {
+    target: 'window',
+    passive: true,
+  })
+  handleNotificationClear() {
+    this.queueDismissed = [...this.queue]
+    this.queue = []
+    this.currentNotification = undefined
+    this.fadeoutTimeouts.forEach((timeoutId) => clearTimeout(timeoutId))
+    this.fadeoutTimeouts.push(
+      setTimeout(() => {
+        this.queueDismissed = []
+      }, FADE_TRANSITION_DURATION)
+    )
+  }
+
+  disconnectedCallback() {
+    clearTimeout(this.dismissTimeout)
+    this.fadeoutTimeouts.forEach((timeoutId) => clearTimeout(timeoutId))
+  }
+
+  render() {
+    const dismissIcon = `
+      <svg
+        width="24"
+        height="24"
+        viewBox="0 0 24 24"
+        fill="none"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <title>Dismiss</title>
+        <path
+          d="M6 6L18 18"
+          stroke="currentColor"
+          stroke-width="3"
+          stroke-linecap="round"
+          stroke-linejoin="round"
+        />
+        <path
+          d="M6 18L18 6"
+          stroke="currentColor"
+          stroke-width="3"
+          stroke-linecap="round"
+          stroke-linejoin="round"
+        />
+      </svg>
+    `
+
+    let cl = 'ld-notification'
+    cl += ` ld-notification--${this.placement}`
+
+    return (
+      <Host class={cl} role="region" aria-label="Notifications">
+        {this.queue.map((notification, index) => (
+          <div
+            class={`ld-notification__item ld-notification__item--${notification.type}`}
+            key={index}
+          >
+            <div
+              class="ld-notification__item-content"
+              innerHTML={notification.content}
+              role={notification.type === 'alert' ? 'alert' : 'status'}
+            ></div>
+            <button
+              class="ld-notification__btn-dismiss"
+              onClick={this.handleNotificationDismiss.bind(this)}
+              innerHTML={dismissIcon}
+            ></button>
+          </div>
+        ))}
+        {this.queueDismissed.map((notification, index) => (
+          <div
+            aria-hidden="true"
```

Comment:
Well, it probably makes no big difference. It's more a definition thing: When can you consider an element as gone for assistive technology? As soon as it starts fading away? Or as soon as it ends fading away? Or even somewhere in between. I decided for as soon as it starts fading away, because when the user clicks the dismiss button, he expects to get rid of the notification immediately.

Reply:
I understand but would also think of this as a little over-engineered, tbh. =)